chainio/clients/elcontracts/reader.go

```diff
@@ -73,6 +74,7 @@ type ELChainReader struct {
 // forces EthReader to implement the chainio.Reader interface
 var _ ELReader = (*ELChainReader)(nil)
 
+// TODO(madhur): make this private. All clients should use build functions
```

Comment:
How about we just deprecate it and eventually get rid of this function?

Reply:
discussed - we don't want to - we will have logger.With(component tag)